crates/sui-types/src/crypto.rs

```diff
@@ -478,6 +480,38 @@ where
     (kp.public().into(), kp)
 }
 
+const DEFAULT_DERIVATION_PATH_PREFIX: &str = "m/44'/784'/0'/0/";
+
+pub fn derive_key_pair_at_index(
+    seed: &[u8],
+    index: u32,
+    key_scheme: &SignatureScheme,
+) -> Result<(SuiAddress, SuiKeyPair), SuiError> {
+    match key_scheme {
+        SignatureScheme::Secp256k1 => {
```

Comment:
i added secp256k1 derivation here for completeness, whether we want to support this under the same derivation path is in question.

Reply:
discussed offline with @kchalkias. we can support both with a 5 part derivation path to distinguish, see:

```
/// Ed25519 follows SLIP-0010 using hardened path: m/44'/784'/0'/0'/{index}'
/// Secp256k1 follows BIP-32 using path where the first 3 levels are hardened: m/44'/784'/1'/0/{index}
```

Reply:
Can you elaborate why we need this distinction?
SLIP-10 allows using the same derivation path for different key types since the first derivation (from the seed) considers the key type/curve. In addition, many wallets use BIP44 and then we might break them if we don't allow them to change the account level, etc.

Reply:
@benr-ml can you provide some reference EIP or doc on a concrete example on how others are doing it? That would be very helpful to not divert from industry standards

Reply:
It's not that common to support more than one signature scheme. Polkadot for example [supports](https://wiki.polkadot.network/docs/learn-accounts#portability) three signature schemes, and then multi-chain wallets like Ledger usually choose to work with only one of them.
Even wallets support SLIP-10, the might have per chain limitations - e.g., Trezor is [using](https://github.com/trezor/trezor-firmware/blob/master/docs/misc/coins-bip44-paths.md#notes) SLIP-10 with a path limitation per chain, and IIUC divert a little from the [registered](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) list of coins.

Specifically re my comment above - If we are going to use SLIP-10, then I don't think we need the limitation on the levels below 784'. That will help us be more consistent with existing ECDSA wallets (following BIP44).

Reply:
good point - this command should instead take in the full derivation path as a option, and no limitation after 784'. 

if we decide to support different signature scheme (not for ledger, but for our own cli paper wallet), we should perhaps use the purpose node instead of the account node. 

https://gist.github.com/miguelmota/034c81886bd23b25c64ec5eda9251641

Reply:
use p = 54 for secp256k1